src/mlpack/core/data/load_impl.hpp

```diff
@@ -113,7 +113,8 @@ bool Load(const std::string& filename,
 
   if (extension == "csv" || extension == "tsv")
   {
-    loadType = arma::diskio::guess_file_type(stream);
+    // In latest Armadillo-9.850.1 guess_file_type is declared arma_deprecated.
+    loadType = arma::diskio::guess_file_type_internal(stream);
```

Comment:
Hey @bisakhmondal thanks for contributing :+1:  .  Linux and windows Builds are failing because they are using 8.400.0 of armadillo and this version does not have `arma::diskio::guess_file_type_internal` Only macos builds are passing because they are using updated version of armadillo. May be you should try to update the pr So, that builds will use only updated version of armadillo

Reply:
Exactly @himanshupathak21061998 . Working on that. Thanks.

Reply:
> Exactly @himanshupathak21061998 . Working on that. Thanks.

```
mlpack has the following dependencies:

  Armadillo     >= 8.400.0
  Boost (program_options, math_c99, unit_test_framework, serialization,
         spirit)
  CMake         >= 3.3.2
  ensmallen     >= 2.10.0
```

We have to support both, may be you can just use version and then just use different function based on that.

Reply:
> Exactly @himanshupathak21061998 . Working on that. Thanks.

I was thinking about my wrong response, that updating a dependency for a single deprecated package is baseless. Thanks @jeffin143, based on version info we can call different function.

Reply:
I think my message is wrongly interpreted sorry about my bad english @jeffin143